src/jest-extensions.js

```diff
@@ -0,0 +1,102 @@
+import {matcherHint, printReceived, printExpected} from 'jest-matcher-utils' //eslint-disable-line import/no-extraneous-dependencies
+
+function getDisplayName(subject) {
+  if (subject && subject.constructor) {
+    return subject.constructor.name
+  } else {
+    return typeof subject
+  }
+}
+
+const assertMessage = (assertionName, message, received, expected) => () =>
+  `${matcherHint(`${assertionName}`, 'received', '')} \n${message}: ` +
+  `${printExpected(expected)} \nReceived: ${printReceived(received)}`
+
+const extensions = {
+  toBeInTheDOM(received) {
+    getDisplayName(received)
+    if (received) {
+      return {
+        message: () =>
+          `${matcherHint(
+            '.not.toBeInTheDOM',
+            'received',
+            '',
+          )} Expected the element not to be present` +
+          `\nReceived : ${printReceived(received)}`,
+        pass: true,
+      }
+    } else {
+      return {
+        message: () =>
+          `${matcherHint(
+            '.toBeInTheDOM',
+            'received',
+            '',
+          )} Expected the element to be present` +
+          `\nReceived : ${printReceived(received)}`,
+        pass: false,
+      }
+    }
+  },
+
+  toHaveTextContent(htmlElement, checkWith) {
+    if (!(htmlElement instanceof HTMLElement))
+      throw new Error(
+        `The given subject is a ${getDisplayName(
+          htmlElement,
+        )}, not an HTMLElement`,
+      )
+
+    const textContent = htmlElement.textContent
+    const pass = textContent === checkWith
```

Comment:
Could we extend this to use the `matcher` function that's in the `queries.js` file? Perhaps we should move that `matcher` function to a `utils` file.

Using the `matcher` function would allow the `toHaveTextContent` to use a regex/case insensitive substring/function which I think would make this more powerful.

Reply:
Yes, thats a good idea. Can take care of that, will move `matcher` to `utils` folder. Import `matcher` in `queries.js` and `jest-extension.js` files to use them. So fundamentally the call would be:

```
const pass = textContent === matches(textContent,htmlElement,checkWith)
```